string source
string destination
string submap_id
